### PR TITLE
Manage products - to be merged after Single Product merge

### PR DESCRIPTION
**Pull request by rfmcmillan**
This is still missing the secret file for Ding's map so the web site won't run, but here are the updates:
- list products 
- add a product 
- edit a product
- the products admin pages are all hooked up to the redux store.

### CODE REVIEW

**Review by alejvillanueva — APPROVED** [review]
Looks great! There is already a singleProduct think and action in the store. I can refactor this code when I add to it.

File: client/components/ManageProducts/EditProduct.js
Comment (alejvillanueva) [inline client/components/ManageProducts/EditProduct.js:143]: Just a general note, again something I can go in and edit when I add to this but reqCountry wouldn't work here because its an included model. You would need to do something like product.country.

Comment (alejvillanueva) [reply client/components/ManageProducts/EditProduct.js:143]: To edit country, it might be easier to do a drop down menu so we have all the countryIds since that's was is on the Product model.

Comment (alejvillanueva) [reply client/components/ManageProducts/EditProduct.js:143]: Instead of passing reqCountry, it should be countryId. Again, I can change al this when I add all the Category stuff to it.

Comment (rfmcmillan) [reply client/components/ManageProducts/EditProduct.js:143]: Thanks, Alejandra. I'll make a note to add the dropdown in. And I should have left more of an explanation on why I used 'reqCountry' instead of 'country'. When I used 'country' in the CreateProduct component's state, that didn't work because the post route for countries was using 'location '. Then I tried to use location, but that didn't work because this.state.location was providing the current url instead of the country's name. I hesitated to change anything up inside the route, but in the end it seemed like just changing the naming of the variable inside the products post route from 'location' to 'reqCountry' was the lowest-impact way to get the post request to go through and get a new product created, rather than messing with anything else within the post route. The product creation was working and all the tests were passing so I figured I didn't throw anything off by changing the naming a bit. But definitely let me know if you still think this will create issues.

Comment (rfmcmillan) [reply client/components/ManageProducts/EditProduct.js:143]: And I'll wait until your PR is merged before I merge this

Comment (alejvillanueva) [reply client/components/ManageProducts/EditProduct.js:143]: So this will work for POST creations because of the way we have the route set up but  it wouldn't work for the Put request since the put route isnt set up that way. If you have a drowndown menu with all the countries already there, instead of reqCountry, we can send in the countryId and it would work for both (you'd just need to change the post route a tiny bit) or you can change the put route to work the way the post route works. This will definetely break some test specs though;

**Pull request merged**
